include/libcyphal/presentation/common_helpers.hpp

```diff
@@ -0,0 +1,69 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_PRESENTATION_COMMON_HELPERS_HPP_INCLUDED
+#define LIBCYPHAL_PRESENTATION_COMMON_HELPERS_HPP_INCLUDED
+
+#include "libcyphal/transport/scattered_buffer.hpp"
+
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <nunavut/support/serialization.hpp>
+
+#include <cstdint>
+#include <memory>
+
+namespace libcyphal
+{
+namespace presentation
+{
+
+/// Internal implementation details of the Presentation layer.
+/// Not supposed to be used directly by the users of the library.
+///
+namespace detail
+{
+
+using DeserializationFailure = cetl::variant<MemoryError, nunavut::support::Error>;
+
+template <typename Message>
```

Comment:
Interesting, we seem to be one template specialization away from closing this todo.